### PR TITLE
refactor and migrate to monolithic architecture

### PR DESCRIPTION
**Pull request by tanlethanh**
- [x] migrate engine
- [x] migrate and refactor authentication
- [x] remove deprecated code
- [x] apply testing for engine
- [x] fix the breaking things from previous migration
- [x] remove ioc
- [x] refactor solana runner
     - [x] assets management
     - [ ] realtime watching
     - [ ] persist assets
     - [ ] persist history 
- [ ] refactor sui runner
- [ ] refactor aptos runner

### CODE REVIEW

File: apps/wallet/src/features/Widget/BuiltInNetwork/AptosTokensTab/PendingTokens.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 import { Image, StyleSheet } from 'react-native';
+import type { AptosPendingToken } from '@walless/core';
 import { shortenAddress } from '@walless/core';
-import type { AptosPendingToken } from '@walless/engine';
-import { aptosState } from '@walless/engine';
 import { Button, Text, View } from '@walless/gui';
 import type { aptosHandler } from '@walless/kernel';
 import { RequestType } from '@walless/messaging';
 import { showRequirePasscodeModal } from 'modals/RequirePasscode';
+import { aptosState } from 'state/assets';
```
Comment (fluctlight-kayaba) [inline apps/wallet/src/features/Widget/BuiltInNetwork/AptosTokensTab/PendingTokens.tsx:9]: use one single convention, sometimes import from `state` this place use `state/assets`

File: apps/wallet/src/screens/Authentication/Login/index.tsx [hunk 2]
```diff
@@ -1,33 +1,76 @@
...
-		useSnapshot(appState);
+	const { invitationCode, config } = useSnapshot(appState);
 	const containerStyle = {
 		paddingTop: insets.top,
 		paddingBottom: insets.bottom,
 	};
-
 	const logoSize = 120;
 
+	const handleGoogleSignIn = async () => {
```
Comment (fluctlight-kayaba) [inline apps/wallet/src/screens/Authentication/Login/index.tsx:31]: should not instantiate method this way, if we must manage it under `useCallback`, or put is as file-scoped level function.

File: apps/wallet/src/state/bootstrap.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	appState.remoteConfig = loadRemoteConfig();
+
+	await Promise.all([
+		configure(storage).then(async () => {
+			if (defaultEngine) return;
+			const engine = await createEngine(storage);
+			await registerNetworkRunners(engine);
+			setDefaultEngine(engine);
+			engine.start();
+		}),
```
Comment (fluctlight-kayaba) [inline apps/wallet/src/state/bootstrap.ts:46]: promote this method to local file-scoped function, which keep this block lean

File: apps/wallet/src/utils/auth/helpers.ts [hunk 4]
```diff
@@ -1,20 +1,17 @@
 import type { Config, UserProfile } from '@walless/core';
+import type { WalletInvitation } from '@walless/graphql';
+import { queries } from '@walless/graphql';
 import type { SettingDocument } from '@walless/store';
+import type { UserAuth } from 'utils/firebase';
```
Comment (fluctlight-kayaba) [inline apps/wallet/src/utils/auth/helpers.ts:5]: UserAuth sound like an action than a type, proposal: `FirebaseUser`

File: apps/wallet/src/utils/auth/signInGoogle.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import Config from 'react-native-config';
+import { GoogleSignin } from '@react-native-google-signin/google-signin';
+import { CustomAuth } from '@toruslabs/customauth-react-native-sdk';
+
+import { auth } from './../firebase/index';
```
Comment (fluctlight-kayaba) [inline apps/wallet/src/utils/auth/signInGoogle.ts:5]: just './../firebase'

**Review by fluctlight-kayaba — APPROVED** [review]

File: apps/wallet/src/engine/runners/solana/tokens.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
+import { TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import type { TokenAmount } from '@solana/web3.js';
+import type { Connection } from '@solana/web3.js';
+import type { PublicKey } from '@solana/web3.js';
```
Comment (FME849) [inline apps/wallet/src/engine/runners/solana/tokens.ts:4]: is this typo/auto-complete error?

File: apps/wallet/src/engine/utils.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import { type Endpoint } from '@walless/core';
+import type { Database } from '@walless/store';
+import { type EndpointsDocument } from '@walless/store';
+
+export const getEndpoints = async (storage: Database) => {
+	let endpoints = (await storage.safeGet('endpoints')) as EndpointsDocument;
```
Comment (FME849) [inline apps/wallet/src/engine/utils.ts:6]: check if this syntax is possible to solve type problem: 
```typescripts
let endpoints = await storage.safeGet<EndpointsDocument>('endpoints');
```

File: apps/wallet/src/state/app.ts [hunk 8]
```diff
@@ -1,13 +1,44 @@
...
+	invitationCode?: string;
+	profile: UserProfile;
+	config: Config;
+	remoteConfig: RemoteConfig;
+	endpoints: EndpointMap;
+	jwtAuth?: string;
+}
+
+export const appState = proxy<AppState>({
+	version: '1.0.0',
```
Comment (FME849) [inline apps/wallet/src/state/app.ts:33]: check if app version is available?

File: apps/wallet/src/utils/auth/helpers.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+		profileImage: user.photoURL as never,
+	};
+};
+
+export const setProfile = async (profile: UserProfile) => {
+	await storage.upsert<SettingDocument>('settings', async (doc) => {
+		doc.type = doc.type || 'Setting';
+		doc.profile = profile;
+		doc.config = doc.config || {
+			version: '1.0.0',
```
Comment (FME849) [inline apps/wallet/src/utils/auth/helpers.ts:28]: check if app version is available to extract?

File: apps/wallet/src/utils/auth/helpers.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+			hideBalance: true,
+			latestLocation: '/',
```
Comment (FME849) [inline apps/wallet/src/utils/auth/helpers.ts:30]: can we remove this hardcode when initializing?

File: apps/wallet/browser/kernel/utils/storage.ts
Comment (FME849) [inline apps/wallet/browser/kernel/utils/storage.ts:1]: did you unify interface among web, ext and mobile?

File: apps/wallet/browser/kernel/utils/helper.ts [hunk 11]
```diff
@@ -13,11 +17,37 @@ export const addExtensionsById = async (id: string) => {
...
 };
 
 export const checkInstalledExtensionById = async (id: string) => {
-	const extension = await modules.storage.safeGet<ExtensionDocument>(id);
+	const extension = await storage.safeGet<ExtensionDocument>(id);
 
 	return extension ? true : false;
 };
+
+export const settings = {
```
Comment (dev-tronghieu) [inline apps/wallet/browser/kernel/utils/helper.ts:29]: this exported `settings` var name is not clear to me

File: apps/wallet/src/modals/Receive/index.tsx [hunk 12]
```diff
@@ -26,17 +25,15 @@ export interface ReceiveModalContext {
...
-			walletList.push({
-				network: key.network.charAt(0).toUpperCase() + key.network.slice(1),
-				networkIcon: networkInfo?.icon ?? { uri: '/img/...' },
-				address: key._id,
-			});
+	keys.forEach((key) => {
+		const networkInfo = getNetworkInfo(key.network);
+		walletList.push({
+			network: key.network.charAt(0).toUpperCase() + key.network.slice(1),
+			networkIcon: networkInfo?.icon ?? { uri: '/img/...' },
```
Comment (dev-tronghieu) [inline apps/wallet/src/modals/Receive/index.tsx:35]: `/img/...` looks weird

File: apps/wallet/src/screens/Authentication/Login/index.tsx [hunk 13]
```diff
@@ -1,33 +1,35 @@
-import type { FC } from 'react';
+import { type FC, useState } from 'react';
```
Comment (dev-tronghieu) [inline apps/wallet/src/screens/Authentication/Login/index.tsx:1]: separate `import type {}` and `import {}`

File: apps/wallet/src/utils/constants.ts [hunk 14]
```diff
@@ -1,5 +1,25 @@
+import { type Config, type RemoteConfig } from '@walless/core';
```
Comment (dev-tronghieu) [inline apps/wallet/src/utils/constants.ts:1]: check this import

File: apps/wallet/src/utils/device/device.ext.ts
Comment (dev-tronghieu) [inline apps/wallet/src/utils/device/device.ext.ts:1]: should name it `index.ext.ts`

File: apps/wallet/browser/kernel/utils/helper.ts (hunk 11 above)
Comment (tanlethanh) [reply apps/wallet/browser/kernel/utils/helper.ts:29]: It was a mock setting for a long time, I just keeping it

**Review by ltminhthu — APPROVED** [review]

**Review by fluctlight-kayaba — APPROVED** [review]

**Pull request merged**